yeat/Snakefile

```diff
@@ -100,19 +101,17 @@ rule downsample:
         base_count = qcdata["summary"]["after_filtering"]["total_bases"]
         read_count = qcdata["summary"]["after_filtering"]["total_reads"]
         avg_read_length = base_count / read_count
-        print(f"average read length: {avg_read_length}")
-        if params.coverage == 150:
-            print(f"default coverage: {params.coverage}")
-        else:
-            print(f"custom coverage: {params.coverage}")
         if params.downsample == 0:
             down = int((genome_size * params.coverage) / (2 * avg_read_length))
-            print(f"auto down: {down}")
         else:
             down = params.downsample
-            print(f"custom down: {down}")
-        shell("seqtk sample {input.read1} {down} > seq/downsample/{wildcards.sample}.R1.fq")
-        shell("seqtk sample {input.read2} {down} > seq/downsample/{wildcards.sample}.R2.fq")
+        seed = randint(1, 2**16-1)
+        print(f"[yeat] average read length: {avg_read_length}")
+        print(f"[yeat] target depth of coverage: {params.coverage}x")
+        print(f"[yeat] number of reads to sample: {down}")
+        print(f"[yeat] random seed for sampling: {seed}")
+        shell("seqtk sample -s {seed} {input.read1} {down} > seq/downsample/{wildcards.sample}.R1.fq")
+        shell("seqtk sample -s {seed} {input.read2} {down} > seq/downsample/{wildcards.sample}.R2.fq")
```

Comment:
The two `seqtk sample` commands need the same seed so that the "random" subset of reads chosen from each file is the same. But in routine use this should really be a randomly (or at least arbitrarily) chosen number. We'll print it out here for debugging purposes, in case a particular seed results in bad behavior that needs to be reproduced.

Reply:
Good to know!